Cannot find module 'json-schema'

Hi there,
I found an issue of @typescript-eslint/eslint-plugin or @typescript-eslint/parser 2.32.0.

<!--
If you have a problem with a specific rule, please begin your issue title with [rulename] to make it easier to search for.
I.e. "[no-unused-vars] False positive when fooing the bar"

Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
Are you opening an issue because the rule you're trying to use is not found?
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨
1) Check the releases log: https://github.com/typescript-eslint/typescript-eslint/releases
    -  If the rule isn't listed there, then chances are it hasn't been released to the main npm tag yet.
2) Try installing the `canary` tag: `npm i @typescript-eslint/eslint-plugin@canary`.
    - The canary tag is built for every commit to master, so it contains the bleeding edge build.
3) If ESLint still can't find the rule, then consider reporting an issue.
-->

<!--
Make sure you read through our FAQ before posting.
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md
-->

**Repro**
```
$ npx eslint --ext .js,.ts .
```

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.
-->

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ["<setting>"]
  }
}
```

```TS
// your repro code case
```

**Expected Result**

**Actual Result**
```
Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js » ./index.js#overrides[0] » ./rules/typescript': Cannot find module 'json-schema'
Require stack:
- /Users/koooge/eslint-config-foo/node_modules/@typescript-eslint/experimental-utils/dist/json-schema.js
- /Users/koooge/eslint-config-foo/node_modules/@typescript-eslint/experimental-utils/dist/index.js
- /Users/koooge/eslint-config-foo/node_modules/@typescript-eslint/eslint-plugin/dist/rules/adjacent-overload-signatures.js
- /Users/koooge/eslint-config-foo/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
- /Users/koooge/eslint-config-foo/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
- /Users/koooge/eslint-config-foo/node_modules/eslint/lib/cli-engine/config-array-factory.js
- /Users/koooge/eslint-config-foo/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js
- /Users/koooge/eslint-config-foo/node_modules/eslint/lib/cli-engine/cli-engine.js
- /Users/koooge/eslint-config-foo/node_modules/eslint/lib/cli-engine/index.js
- /Users/koooge/eslint-config-foo/node_modules/eslint/lib/cli.js
- /Users/koooge/eslint-config-foo/node_modules/eslint/bin/eslint.js
Referenced from: /Users/koooge/eslint-config-foo/rules/typescript.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (/Users/koooge/eslint-config-foo/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (/Users/koooge/eslint-config-foo/node_modules/@typescript-eslint/experimental-utils/dist/json-schema.js:7:10)
    at Module._compile (/Users/koooge/eslint-config-foo/node_modules/v8-compile-cache/v8-compile-cache.js:194:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! eslint-config-foo@0.0.1 test: `eslint --ext .js,.ts .`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the eslint-config-foo@0.0.1 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/koooge/.npm/_logs/2020-05-11T17_59_00_930Z-debug.log
```

**Additional Info**

<!--
Did eslint throw an exception?

Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug
-->

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.32.0` |
| `@typescript-eslint/parser`        | `2.32.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.3` |
| `npm`                              | `6.14.5` |

--- a/packages/experimental-utils/src/json-schema.ts
+++ b/packages/experimental-utils/src/json-schema.ts
@@ -1,2 +1,24 @@
+// Note - @types/json-schema@7.0.4 added some function declarations to the type package
+// If we do export *, then it will also export these function declarations.
+// This will cause typescript to not scrub the require from the build, breaking anyone who doesn't have it as a dependency
+
 // eslint-disable-next-line import/no-extraneous-dependencies
-export * from 'json-schema';
+export {
+  JSONSchema4,
+  JSONSchema4Type,
+  JSONSchema4TypeName,
+  JSONSchema4Version,
+  JSONSchema6,
+  JSONSchema6Definition,
+  JSONSchema6Type,
+  JSONSchema6TypeName,
+  JSONSchema6Version,
+  JSONSchema7,
+  JSONSchema7Array,
+  JSONSchema7Definition,
+  JSONSchema7Type,
+  JSONSchema7TypeName,
+  JSONSchema7Version,
+  ValidationError,
+  ValidationResult,
+} from 'json-schema';

